dff/core/pipeline/pipeline/pipeline.py

```diff
@@ -189,6 +189,7 @@ async def _run_pipeline(self, request: Any, ctx_id: Optional[Hashable] = None) -
         self.context_storage[ctx_id] = ctx
         return ctx
 
+    # TODO: test this func by pytest
```

Comment:
I'm leaving this TODO without testing for two reasons:
1. `Pipeline.run()` is a completely interactive-mode function, it starts an infinite loop, that is not suitable for testing.
2. Tests in `test_messenger_interface.py` file (coming in next commit) completely repeat it's functionality but are finite.